src/main/java/com/iabtcf/FieldDefs.java

```diff
@@ -0,0 +1,427 @@
+    CORE_VERSION(6, 0),
+    CORE_CREATED(36),
+    CORE_LAST_UPDATED(36),
+    CORE_CMP_ID(12),
+    CORE_CMP_VERSION(12),
+    CORE_CONSENT_SCREEN(6),
+    CORE_CONSENT_LANGUAGE(12),
+    CORE_VENDOR_LIST_VERSION(12),
+    CORE_TCF_POLICY_VERSION(6),
+    CORE_IS_SERVICE_SPECIFIC(1),
+    CORE_USE_NON_STANDARD_STOCKS(1),
+    CORE_SPECIAL_FEATURE_OPT_INS(12),
+    CORE_PURPOSES_CONSENT(24),
+    CORE_PURPOSES_LI_TRANSPARENCY(24),
+    CORE_PURPOSE_ONE_TREATMENT(1),
+    CORE_PUBLISHER_CC(12),
+    CORE_VENDOR_MAX_VENDOR_ID(16),
+    CORE_VENDOR_IS_RANGE_ENCODING(1),
+    CORE_VENDOR_BITRANGE_FIELD(
+            BitRangeFieldUtils.lengthSupplier(CORE_VENDOR_IS_RANGE_ENCODING, CORE_VENDOR_MAX_VENDOR_ID)),
+    CORE_VENDOR_LI_MAX_VENDOR_ID(16),
+    CORE_VENDOR_LI_IS_RANGE_ENCODING(1),
+    CORE_VENDOR_LI_BITRANGE_FIELD(
+            BitRangeFieldUtils.lengthSupplier(CORE_VENDOR_LI_IS_RANGE_ENCODING, CORE_VENDOR_LI_MAX_VENDOR_ID)),
+    CORE_NUM_PUB_RESTRICTION(12),
+    CORE_PUB_RESTRICTION_ENTRY(
+            PublisherRestrictionUtils.lengthSupplier(CORE_NUM_PUB_RESTRICTION)),
+
+    OOB_SEGMENT_TYPE(3, 0),
+
+    // disallowed vendor fields
+    DV_MAX_VENDOR_ID(16, OOB_SEGMENT_TYPE),
+    DV_IS_RANGE_ENCODING(1),
+    DV_VENDOR_BITRANGE_FIELD(
+            BitRangeFieldUtils.lengthSupplier(DV_IS_RANGE_ENCODING, DV_MAX_VENDOR_ID)),
+
+    // allowed vendor fields
+    AV_MAX_VENDOR_ID(16, OOB_SEGMENT_TYPE),
+    AV_IS_RANGE_ENCODING(1),
+    AV_VENDOR_BITRANGE_FIELD(
+            BitRangeFieldUtils.lengthSupplier(AV_IS_RANGE_ENCODING, AV_MAX_VENDOR_ID)),
+
+    // publisher purposes transparency and consent
+    PPTC_SEGMENT_TYPE(3, 0),
+    PPTC_PUB_PURPOSES_CONSENT(24),
+    PPTC_PUB_PURPOSES_LI_TRANSPARENCY(24),
+    PPTC_NUM_CUSTOM_PURPOSES(6),
+    PPTC_CUSTOM_PURPOSES_CONSENT(new LengthSupplier() {
+        @Override
+        public Integer apply(ByteBitVector t) {
+            return Integer.valueOf(t.readBits6(PPTC_NUM_CUSTOM_PURPOSES.getOffset(t)));
+        }
+
+        @Override
+        public boolean isDynamic() {
+            return true;
+        }
+    }),
+    PPTC_CUSTOM_PURPOSES_LI_TRANSPARENCY(new LengthSupplier() {
+        @Override
+        public Integer apply(ByteBitVector t) {
+            // same length as PPTC_CUSTOM_PURPOSES_CONSENT
+            return PPTC_CUSTOM_PURPOSES_CONSENT.getLength(t);
+        }
+
+        @Override
+        public boolean isDynamic() {
+            return true;
+        }
+    }),
+
+    // range entry, only field lengths are supported
+    NUM_ENTRIES(12, OffsetSupplier.NOT_SUPPORTED),
+    IS_A_RANGE(1, OffsetSupplier.NOT_SUPPORTED),
+    START_OR_ONLY_VENDOR_ID(16, OffsetSupplier.NOT_SUPPORTED),
+    END_VENDOR_ID(16, OffsetSupplier.NOT_SUPPORTED),
+
+    // publish restriction fields, only field lengths are supported
+    PURPOSE_ID(6, OffsetSupplier.NOT_SUPPORTED),
+    RESTRICTION_TYPE(2, OffsetSupplier.NOT_SUPPORTED),
+
+    // v1 fields
+    V1_VERSION(6, 0),
```

Comment:
These V1 fields are only used in tests. I'd remove for now until we use them in the V1 implementation.

Reply:
i've modified the V1 implementation to use FieldDefs in a recent commit in this PR.